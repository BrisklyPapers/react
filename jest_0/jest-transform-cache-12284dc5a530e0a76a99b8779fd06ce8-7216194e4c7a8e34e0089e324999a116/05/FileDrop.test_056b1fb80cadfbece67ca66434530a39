'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _FileDrop = require('../../src/components/FileDrop.js');var _FileDrop2 = _interopRequireDefault(_FileDrop);
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('components/FileDrop', function () {
    it('should render correctly', function () {
        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: [] }));

        expect(fileDrop.toJSON()).toMatchSnapshot();
    });

    it('should change style onDragOver and onDragEnd', function () {
        var event = {
            preventDefault: function preventDefault() {
            } };


        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: [] }));

        var tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();

        tree.children[0].props.onDragOver(event);
        tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();

        tree.children[0].props.onDragEnd(event);
        tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should change style onDragOver and onDragLeave', function () {
        var event = {
            preventDefault: function preventDefault() {
            } };


        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: [] }));

        var tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();

        tree.children[0].props.onDragOver(event);
        tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();

        tree.children[0].props.onDragLeave(event);
        tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should add and return dropped files', function () {
        var event = {
            preventDefault: function preventDefault() {
            },
            dataTransfer: {
                files: [
                { name: 'file1' },
                { name: 'file2' }] } };




        var files = [];
        var dropFiles = function dropFiles(droppedFiles) {
            files = droppedFiles;
        };

        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: dropFiles, files: files }));


        tree.children[0].props.onDrop(event);
        tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();

        expect(files).toEqual([{ file1: file1 }, { file2: file2 }]);
    });
});