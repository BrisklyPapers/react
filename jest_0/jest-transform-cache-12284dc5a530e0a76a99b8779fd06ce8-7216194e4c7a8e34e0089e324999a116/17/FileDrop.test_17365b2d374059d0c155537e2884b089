'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _FileDrop = require('../../src/components/FileDrop.js');var _FileDrop2 = _interopRequireDefault(_FileDrop);
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);
var _enzyme = require('enzyme');
var _sinon = require('sinon');function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('components/FileDrop', function () {
    it('should render correctly', function () {
        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: [] }));

        expect(fileDrop.toJSON()).toMatchSnapshot();
    });

    it('should change style onDragOver and onDragEnd', function () {
        var event = {
            preventDefault: function preventDefault() {
            } };


        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: [] }));

        var tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();

        var divBox = tree.children[0];
        expect(divBox.props.id).toEqual('fileDrop');

        divBox.props.onDragOver(event);
        expect(fileDrop.toJSON()).toMatchSnapshot();

        divBox.props.onDragEnd(event);
        expect(fileDrop.toJSON()).toMatchSnapshot();
    });

    it('should change style onDragOver and onDragLeave', function () {
        var event = {
            preventDefault: function preventDefault() {
            } };


        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: [] }));

        var tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();

        tree.children[0].props.onDragOver(event);
        expect(fileDrop.toJSON()).toMatchSnapshot();

        tree.children[0].props.onDragLeave(event);
        expect(fileDrop.toJSON()).toMatchSnapshot();
    });

    it('should add and return dropped files', function () {
        var event = {
            preventDefault: function preventDefault() {
            },
            dataTransfer: {
                files: [
                { name: 'file1' },
                { name: 'file2' }] } };




        var files = [];
        var dropFiles = function dropFiles(droppedFiles) {
            files = droppedFiles;
        };

        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: dropFiles, files: files }));


        var tree = fileDrop.toJSON();
        tree.children[0].props.onDrop(event);
        expect(fileDrop.toJSON()).toMatchSnapshot();

        expect(files).toEqual([{ name: 'file1' }, { name: 'file2' }]);
    });

    it('should add and return dropped files if files are items', function () {
        var event = {
            preventDefault: function preventDefault() {
            },
            dataTransfer: {
                items: [
                { name: 'file1' },
                { name: 'file2' }] } };




        var files = [];
        var dropFiles = function dropFiles(droppedFiles) {
            files = droppedFiles;
        };

        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: dropFiles, files: files }));


        var tree = fileDrop.toJSON();
        tree.children[0].props.onDrop(event);
        expect(fileDrop.toJSON()).toMatchSnapshot();

        expect(files).toEqual([{ name: 'file1' }, { name: 'file2' }]);
    });

    it('should add and return dropped files if files are added by file input', function () {
        var event = {
            preventDefault: function preventDefault() {
            },
            target: {
                files: [
                { name: 'file1' },
                { name: 'file2' }] } };




        var files = [];
        var dropFiles = function dropFiles(droppedFiles) {
            files = droppedFiles;
        };

        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: dropFiles, files: files }));


        var tree = fileDrop.toJSON();

        var fileInput = tree.children[0].children[3];
        expect(fileInput.props.type).toEqual('file');
        expect(fileInput.props.name).toEqual('files[]');

        fileInput.props.onChange(event);
        expect(fileDrop.toJSON()).toMatchSnapshot();

        expect(files).toEqual([{ name: 'file1' }, { name: 'file2' }]);
    });

    it('should ignore unkown events', function () {
        var event = {
            preventDefault: function preventDefault() {
            } };


        var files = [];
        var dropFiles = function dropFiles(droppedFiles) {
            files = droppedFiles;
        };

        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: dropFiles, files: files }));


        var tree = fileDrop.toJSON();
        tree.children[0].props.onDrop(event);
        expect(fileDrop.toJSON()).toMatchSnapshot();

        expect(files).toEqual([]);
    });

    it('should ignore incomplete dataTransfer events', function () {
        var event = {
            preventDefault: function preventDefault() {
            },
            dataTransfer: {} };


        var files = [];
        var dropFiles = function dropFiles(droppedFiles) {
            files = droppedFiles;
        };

        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: dropFiles, files: files }));


        var tree = fileDrop.toJSON();
        tree.children[0].props.onDrop(event);
        expect(fileDrop.toJSON()).toMatchSnapshot();

        expect(files).toEqual([]);
    });

    it('should display given files', function () {
        var files = [{ name: 'file1' }, { name: 'file2' }];

        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: files }));

        expect(fileDrop.toJSON()).toMatchSnapshot();
    });

    it('should trigger click event on file input', function (done) {
        var fileDrop = (0, _enzyme.mount)(_react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: [] }));
        var clickSpy = (0, _sinon.spy)(fileDrop.instance().inputElement, 'click');
        fileDrop.find('#fileDrop').simulate('click');
        setTimeout(function () {
            expect(clickSpy.callCount).toEqual(1);
            done();
        }, 0);
    });

    it('should preventDefault when calling cancelEvent', function () {
        var fileDrop = (0, _enzyme.mount)(_react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: [] }));
        var event = {};
        event.preventDefault = (0, _sinon.spy)();

        fileDrop.instance().cancelEvent(event);

        expect(event.preventDefault.callCount).toEqual(1);
    });

    it('should ignore preventDefault when calling cancelEvent with incomplete event', function () {
        var fileDrop = (0, _enzyme.mount)(_react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: [] }));
        expect(fileDrop.instance().cancelEvent({})).toEqual(undefined);
    });
});