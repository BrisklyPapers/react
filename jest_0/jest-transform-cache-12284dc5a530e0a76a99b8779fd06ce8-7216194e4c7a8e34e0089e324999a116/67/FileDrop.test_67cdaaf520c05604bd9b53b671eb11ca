'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _FileDrop = require('../../src/components/FileDrop.js');var _FileDrop2 = _interopRequireDefault(_FileDrop);
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('components/FileDrop', function () {
    it('should render correctly', function () {
        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: [] }));

        expect(fileDrop.toJSON()).toMatchSnapshot();
    });

    it('should change style onDragOver', function () {
        var event = {
            preventDefault: function preventDefault() {
            } };


        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: [] }));

        var tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();

        tree.children[0].props.onDragOver(event);
        tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();

        tree.children[0].props.onDragEnd(event);
        tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();

    });
});