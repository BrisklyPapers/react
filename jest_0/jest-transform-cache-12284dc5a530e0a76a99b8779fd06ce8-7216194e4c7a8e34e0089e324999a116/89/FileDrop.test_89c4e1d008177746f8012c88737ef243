'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _FileDrop = require('../../src/components/FileDrop.js');var _FileDrop2 = _interopRequireDefault(_FileDrop);
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('components/FileDrop', function () {
    it('renders correctly', function () {
        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: [] }));

        expect(fileDrop.toJSON()).toMatchSnapshot();
    });
});