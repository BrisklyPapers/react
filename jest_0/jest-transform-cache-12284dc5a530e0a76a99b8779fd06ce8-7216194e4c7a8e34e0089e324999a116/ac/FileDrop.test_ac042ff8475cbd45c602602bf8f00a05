'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _FileDrop = require('../../src/components/FileDrop.js');var _FileDrop2 = _interopRequireDefault(_FileDrop);
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);
var _testUtils = require('react-dom/test-utils');var _testUtils2 = _interopRequireDefault(_testUtils);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('components/FileDrop', function () {
    it('should render correctly', function () {
        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: [] }));

        expect(fileDrop.toJSON()).toMatchSnapshot();
    });

    it('should change style onDragOver', function () {

        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: [] }));

        var tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();

        tree.children[0].props.onDragOver();
        tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();


    });
});