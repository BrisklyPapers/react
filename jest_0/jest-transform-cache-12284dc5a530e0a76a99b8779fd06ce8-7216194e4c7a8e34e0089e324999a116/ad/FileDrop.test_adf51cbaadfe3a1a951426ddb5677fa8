'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _FileDrop = require('../../src/components/FileDrop.js');var _FileDrop2 = _interopRequireDefault(_FileDrop);
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('components/FileDrop', function () {
    it('should render correctly', function () {
        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: [] }));

        expect(fileDrop.toJSON()).toMatchSnapshot();
    });

    it('should change style onDragOver and onDragEnd', function () {
        var event = {
            preventDefault: function preventDefault() {
            } };


        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: [] }));

        var tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();

        var divBox = tree.children[0];
        expect(divBox.props.id).toEqual('fileDrop');

        divBox.props.onDragOver(event);
        expect(fileDrop.toJSON()).toMatchSnapshot();

        divBox.props.onDragEnd(event);
        expect(fileDrop.toJSON()).toMatchSnapshot();
    });

    it('should change style onDragOver and onDragLeave', function () {
        var event = {
            preventDefault: function preventDefault() {
            } };


        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: [] }));

        var tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();

        tree.children[0].props.onDragOver(event);
        expect(fileDrop.toJSON()).toMatchSnapshot();

        tree.children[0].props.onDragLeave(event);
        expect(fileDrop.toJSON()).toMatchSnapshot();
    });

    it('should add and return dropped files', function () {
        var event = {
            preventDefault: function preventDefault() {
            },
            dataTransfer: {
                files: [
                { name: 'file1' },
                { name: 'file2' }] } };




        var files = [];
        var dropFiles = function dropFiles(droppedFiles) {
            files = droppedFiles;
        };

        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: dropFiles, files: files }));


        var tree = fileDrop.toJSON();
        tree.children[0].props.onDrop(event);
        expect(fileDrop.toJSON()).toMatchSnapshot();

        expect(files).toEqual([{ name: 'file1' }, { name: 'file2' }]);
    });

    it('should add and return dropped files if files are items', function () {
        var event = {
            preventDefault: function preventDefault() {
            },
            dataTransfer: {
                items: [
                { name: 'file1' },
                { name: 'file2' }] } };




        var files = [];
        var dropFiles = function dropFiles(droppedFiles) {
            files = droppedFiles;
        };

        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: dropFiles, files: files }));


        var tree = fileDrop.toJSON();
        tree.children[0].props.onDrop(event);
        expect(fileDrop.toJSON()).toMatchSnapshot();

        expect(files).toEqual([{ name: 'file1' }, { name: 'file2' }]);
    });

    it('should add and return dropped files if files are added by file input', function () {
        var event = {
            preventDefault: function preventDefault() {
            },
            target: {
                files: [
                { name: 'file1' },
                { name: 'file2' }] } };




        var files = [];
        var dropFiles = function dropFiles(droppedFiles) {
            files = droppedFiles;
        };

        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: dropFiles, files: files }));


        var tree = fileDrop.toJSON();

        var fileInput = tree.children[0].children[3];
        expect(fileInput.props.type).toEqual('file');
        expect(fileInput.props.name).toEqual('files[]');

        fileInput.props.onChange(event);
        expect(fileDrop.toJSON()).toMatchSnapshot();

        expect(files).toEqual([{ name: 'file1' }, { name: 'file2' }]);
    });

    it('should display given files', function () {
        var files = [{ name: 'file1' }, { name: 'file2' }];

        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: files }));

        expect(fileDrop.toJSON()).toMatchSnapshot();
    });

    it('should trigger input file onClick', function () {
        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: [] }));


        console.log(fileDrop.toJson());
    });
});