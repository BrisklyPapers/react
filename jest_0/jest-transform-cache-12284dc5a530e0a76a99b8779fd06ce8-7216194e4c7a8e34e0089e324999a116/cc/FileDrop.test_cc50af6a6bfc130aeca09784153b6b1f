'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _FileDrop = require('../../src/components/FileDrop.js');var _FileDrop2 = _interopRequireDefault(_FileDrop);
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('components/FileDrop', function () {
    it('should render correctly', function () {
        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: [] }));

        expect(fileDrop.toJSON()).toMatchSnapshot();
    });

    it('should change style onDragOver and onDragEnd', function () {
        var event = {
            preventDefault: function preventDefault() {
            } };


        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: [] }));

        var tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();

        tree.children[0].props.onDragOver(event);
        tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();

        tree.children[0].props.onDragEnd(event);
        tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should change style onDragOver and onDragLeave', function () {
        var event = {
            preventDefault: function preventDefault() {
            } };


        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: [] }));

        var tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();

        tree.children[0].props.onDragOver(event);
        tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();

        tree.children[0].props.onDragLeave(event);
        tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should add and return dropped files', function () {
        var event = {
            preventDefault: function preventDefault() {
            },
            dataTransfer: {
                files: [
                { name: 'file1' },
                { name: 'file2' }] } };




        var files = [];
        var dropFiles = function dropFiles(droppedFiles) {
            files = droppedFiles;
        };

        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: dropFiles, files: files }));


        var tree = fileDrop.toJSON();
        tree.children[0].props.onDrop(event);
        tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();

        expect(files).toEqual([{ name: 'file1' }, { name: 'file2' }]);
    });

    it('should add and return dropped files if files are items', function () {
        var event = {
            preventDefault: function preventDefault() {
            },
            dataTransfer: {
                items: [
                { name: 'file1' },
                { name: 'file2' }] } };




        var files = [];
        var dropFiles = function dropFiles(droppedFiles) {
            files = droppedFiles;
        };

        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: dropFiles, files: files }));


        var tree = fileDrop.toJSON();
        tree.children[0].props.onDrop(event);
        tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();

        expect(files).toEqual([{ name: 'file1' }, { name: 'file2' }]);
    });

    it('should display given files', function () {
        var files = [{ name: 'file1' }, { name: 'file2' }];

        var fileDrop = _reactTestRenderer2.default.create(
        _react2.default.createElement(_FileDrop2.default, { dropFiles: function dropFiles() {}, files: files }));

        var tree = fileDrop.toJSON();
        expect(tree).toMatchSnapshot();
    });
});