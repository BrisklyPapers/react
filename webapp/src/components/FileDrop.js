import React from 'react';
import Radium from 'radium';
import PropTypes from 'prop-types';

class FileDrop extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            dragState: "",
        }
    };

    render() {
        return (
            <div>
                <div
                    onDragOver={ this.dragOver }
                    onDragEnd={ this.dragLeave }
                    onDragLeave={ this.dragLeave }

                    onDrop={ this.onDrop }
                    onClick={ this.onClick }

                    style={[
                        styles.base,
                        styles[this.state.dragState]
                    ]}
                >
                    Drop files here or click to <span id="browse">browse</span>.
                    <input
                        id="fileBox"
                        type="file"
                        name="files[]"
                        multiple

                        style={[styles.fileBox]}

                        ref={input => this.inputElement = input}

                        onChange={ this.onDrop }
                    />
                </div>
                <ul>
                    {this.props.files.map((file) =>
                        <li key={file.name}>{file.name}</li>
                    )}
                </ul>
            </div>
        );
    };

    cancelEvent(e) {
        e = e || window.event; // get window.event if e argument missing (in IE)
        if (e.preventDefault) {
            e.preventDefault();
        }
    };

    dragOver(e) {
        this.cancelEvent(e);
        this.setState({dragState: "hover"});
        return false;
    };

    dragLeave(e) {
        this.cancelEvent(e);
        this.setState({dragState: ""});
        return false;
    };

    onDrop(e) {
        this.dragLeave(e);

        let files = [];
        if (e.dataTransfer) {
            const dt = e.dataTransfer;
            if (dt.files && dt.files.length) {
                files = dt.files
            } else if (dt.items && dt.items.length) {
                files = dt.items
            }
        }  else if (e.target && e.target.files) {
            files = e.target.files
        }

        this.props.dropFiles(files);

        return false;
    };

    onClick(e) {
        this.inputElement.click();
    };
}

FileDrop.propTypes = {
    dropFiles: PropTypes.func.isRequired,
    files: PropTypes.array.isRequired
};

var styles = {
    base: {
        border: "1px dashed #ccc",
        padding: "10px",
        minHeight: "100px"
    },
    hover: {
        border: "2px dashed rgb(0, 188, 212)"
    },
    fileBox: {
        display: "none"
    }
};

FileDrop = Radium(FileDrop);

export default FileDrop;