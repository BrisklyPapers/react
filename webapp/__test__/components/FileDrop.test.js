import React from 'react';
import FileDrop from '../../src/components/FileDrop.js';
import renderer from 'react-test-renderer';

describe('components/FileDrop', () => {
    it('renders correctly', () => {
        const fileDrop = renderer.create(
            <FileDrop dropFiles={() => {}} files={[]} />
        );
        expect(fileDrop.toJSON()).toMatchSnapshot();
    });
});