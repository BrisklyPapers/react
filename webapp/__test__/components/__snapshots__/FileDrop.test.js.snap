// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/FileDrop renders correctly 1`] = `
<div
  data-radium={true}
>
  <div
    data-radium={true}
    onClick={[Function]}
    onDragEnd={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "border": "1px dashed #ccc",
        "minHeight": "100px",
        "padding": "10px",
      }
    }
  >
    Drop files here or click to 
    <span
      id="browse"
    >
      browse
    </span>
    .
    <input
      data-radium={true}
      id="fileBox"
      multiple={true}
      name="files[]"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
  </div>
  <ul
    data-radium={true}
  />
</div>
`;
